apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -194,4 +196,118 @@ defmodule Snitch.Domain.Order.Transitions do
         {:halt, error}
     end)
   end
+
+  # Guard Class to check negative amount is skipped.
+
+  defp calc_payable_amount(order) do
```

Comment:
This should be in `Model.Order`, and name this better, simple `payable_amount/1` works best.